### PR TITLE
add cluster support

### PR DESCRIPTION
**Pull request by MrChengmo**
- Supoort PaddleCloud MPI-PS-CPU training

- Support PaddleCloud K8S-Collective-GPU training

image

### CODE REVIEW

File: core/engine/cluster/cluster.py [hunk 1]
```diff
@@ -55,21 +67,229 @@ def start_master_procs(self):
...
+            "config.output_path", "")
+        if self.cluster_env["OUTPUT_PATH"] == "":
+            warnings.warn(
+                "Job output_path not set! Please check your backend yaml.",
+                category=UserWarning,
+                stacklevel=2)
+
+        # paddle_version
+        self.cluster_env["PADDLE_VERSION"] = self.backend_env.get(
+            "config.paddle_version", "1.7.2")
```
Comment (seiriosPlus) [inline core/engine/cluster/cluster.py:162]: why need this?

File: models/rank/dnn/config.yaml [hunk 2]
```diff
@@ -52,6 +52,7 @@ hyper_parameters:
 
 # select runner by name
 mode: [single_cpu_train, single_cpu_infer]
+# mode: [single_cpu_train, single_cpu_infer]
```
Comment (seiriosPlus) [inline models/rank/dnn/config.yaml:55]: 多余的一行？

File: core/engine/cluster/cluster.py (hunk 1 above)
Comment (MrChengmo) [reply core/engine/cluster/cluster.py:162]: 方便用户指定paddle版本。。。PaddleCloud上只支持到1.7.1

File: models/rank/dnn/config.yaml (hunk 2 above)
Comment (MrChengmo) [reply models/rank/dnn/config.yaml]: 已删除

**Pull request merged**